src/adapter/graphql/src/queries/datasets/adapters/versioned_file.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use kamu::domain;
+
+use super::{FileVersion, VersionedFileEntry, VersionedFileEntryConnection};
+use crate::prelude::*;
+use crate::queries::DatasetRequestStateWithOwner;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug)]
+pub struct VersionedFile {
+    state: DatasetRequestStateWithOwner,
+}
+
+impl VersionedFile {
+    pub fn new(state: DatasetRequestStateWithOwner) -> Self {
+        Self { state }
+    }
+
+    pub async fn get_entry(
+        &self,
+        ctx: &Context<'_>,
+        as_of_version: Option<FileVersion>,
+        as_of_block_hash: Option<Multihash<'static>>,
```

Comment:
Q: How about crossing these two parameters into one enum?

Thus, the user will not be able to pass them both at the same time